### PR TITLE
user customizable all spinners in ingredient class

### PR DESCRIPTION
**Pull request by antarc0y**
- user can add their own category, location, and units in addition to predetermined spinner options

### CODE REVIEW

File: android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java [hunk 1]
```diff
@@ -211,8 +271,35 @@ public void onNothingSelected(AdapterView<?> adapterView) {
              * @param l {@link Long} the row ID of the item that was selected
              */
             public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
-                unitName.setSelection(i);
-                currentIngredient.setUnit(adapterView.getItemAtPosition(i).toString());
+                // if Add unit is selected, set the textbox to visible
+                if (unitAdapter.getItem(i).equals("Add Unit")) {
+                    EditText customUnit = new EditText(getContext());
+                    //customUnit.setVisibility(view.VISIBLE);
```
Comment (yismailuofa) [inline android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java:277]: Delete comment

File: android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java [hunk 2]
```diff
@@ -211,8 +271,35 @@ public void onNothingSelected(AdapterView<?> adapterView) {
...
+                                    unitOptions.add(newUnit);
+                                    unitAdapter.notifyDataSetChanged();
+                                    int j = unitAdapter.getPosition(newUnit);
+                                    unitName.setSelection(j);
+                                }
+                            }).show();
+                }
+                else {
+                    // user didn't select the add custom option
+                    //unitName.setSelection(i);
```
Comment (yismailuofa) [inline android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java:300]: Delete fommwnt

Comment (antarc0y) [reply android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java:300]: what do you mean delete comment

Comment (antarc0y) [reply android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java]: deleted

File: android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java [hunk 3]
```diff
@@ -149,8 +157,33 @@ public void onClick(DialogInterface dialog, int id) {
              */
             @Override
             public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
-                categoryName.setSelection(i);
-                currentIngredient.setCategory(adapterView.getItemAtPosition(i).toString());
+                if (categoryAdapter.getItem(i).equals("Add Category")) {
+                    EditText customCategory = new EditText(getContext());
+                    //customUnit.setVisibility(view.VISIBLE);
```
Comment (yismailuofa) [inline android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java:162]: Should prolly delete the comments from these the other two spinners too lol

File: android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java [hunk 4]
```diff
@@ -179,8 +213,34 @@ public void onNothingSelected(AdapterView<?> adapterView) {
              * @param l {@link Long} the row ID of the item that was selected
              */
             public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
-                locationName.setSelection(i);
-                currentIngredient.setLocation(adapterView.getItemAtPosition(i).toString());
+                if (locationAdapter.getItem(i).equals("Add Location")) {
+                    EditText customLocation = new EditText(getContext());
+                    //customUnit.setVisibility(view.VISIBLE);
```
Comment (yismailuofa) [inline android/301Project/app/src/main/java/com/example/a301project/AddEditIngredientFragment.java:218]: ~

**Review by yismailuofa — APPROVED** [review]

**Pull request merged**